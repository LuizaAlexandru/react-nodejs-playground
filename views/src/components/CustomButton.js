import React, {Component} from 'react';
import PropTypes from 'prop-types';

class CustomButton extends Component {
  constructor(props) {
    super(props);
    this.handleClick = this.handleClick.bind(this);
  }

  render() {
    return <div className="action" onClick={this.handleClick}>{
      this.props.placeholder ?
      <span className="action-placeholder">{this.props.placeholder}</span> : this.props.text
    }</div>
  }

  handleClick(e) {
    e.preventDefault();
    this.props.callback(this.props.id);
  }
}

export default CustomButton;

CustomButton.propTypes = {
  id: PropTypes.string,
  text: PropTypes.string,
  placeholder: PropTypes.string,
  callback: PropTypes.func
}
