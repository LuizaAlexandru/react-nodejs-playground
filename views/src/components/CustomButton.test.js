import React from 'react';
import renderer from 'react-test-renderer';

import CustomButton from './CustomButton.jsx';


test('Initiated CustomButton matches the snapshot', () => {
  const component = renderer.create(<CustomButton />);
  expect(component.toJSON()).toMatchSnapshot();
});

test('action has text', () => {
  let button = mount(
    <CustomButton text="text property" />
  );
  expect(button.text()).toBe('text property');
});

test('handleClickCustomButton preventsDefault event', () => {

  const data = '398456394-865';
  let button = shallow(
    <CustomButton text="text property" id={data} callback={(receivedData) => {
      expect(receivedData).toEqual(data);
    }} />
  );

  let e = {};
  e.preventDefault = () => {
    expect(true).toEqual(true);
  };
  button.simulate('click', e);
})
