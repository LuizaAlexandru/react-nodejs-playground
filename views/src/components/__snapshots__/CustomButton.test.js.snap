// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Initiated CustomButton matches the snapshot 1`] = `
<div
  className="action"
  onClick={[Function]}
/>
`;
